compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -807,9 +816,29 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // In that case, we return early. We also avoid validation in that case,
         // because this is CTFE and the final value will be thoroughly validated anyway.
         let (cleanup, next_block) = match frame.return_to_block {
-            StackPopCleanup::Goto { ret, unwind } => {
-                (true, Some(if unwinding { unwind } else { ret }))
-            }
+            StackPopCleanup::Goto { ret, unwind } => (
+                true,
+                Some(if unwinding {
```

Comment:
Please move this `if` up a bit, this is nesting control flow a bit too deeply. I think putting it in the match would work, like
```rust
match (frame.return_to_block, unwinding) {
  (StackPopCleanup::Goto { ret, _ }, false) => ...
  (StackPopCleanup::Goto { _, unwind }, true) => ...
  (StackPopCleanup::None { cleanup, .. }, _) => ...
}
```